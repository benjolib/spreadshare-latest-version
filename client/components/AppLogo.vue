<template>
  <div :class="$style.logo">
    <h1>
      <a href="https://github.com/wemake-services/wemake-vue-template">
        Spreadshare Boilerplate
      </a>
    </h1>
  </div>
</template>

<style lang="scss" module>
@import '~/scss/variables';
@import '~/scss/mixins';

.logo {
  @include centered();

  a,
  a:visited,
  a:active {
    color: $color-black;
  }
}
</style>
