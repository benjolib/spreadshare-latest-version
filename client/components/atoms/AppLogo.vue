<template>
  <nuxt-link
    to="/"
  >
    <img
      :class="$style.logo"
      src="~assets/logo.svg"
      alt="logo"
    >
  </nuxt-link>
</template>

<style lang="scss" module>
@import '@design';

.logo {
  display: inline-block;
  width: $size-logo;
  vertical-align: middle;
}
</style>
