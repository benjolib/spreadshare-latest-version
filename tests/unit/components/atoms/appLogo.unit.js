// @Flow

import { shallowMount } from '@vue/test-utils'
import AppLogo from '@atoms/AppLogo'

describe('appLogo', () => {
  const wrapper = shallowMount(AppLogo, {
    stubs: ['nuxt-link'],
  })

  describe('unit tests', () => {
    test('is a Vue instance', () => {
      expect(wrapper.isVueInstance()).toBeTruthy()
      expect(wrapper.is(AppLogo)).toBeTruthy()
    })

    test('should have a nuxt-link pointing to /', () => {
      const link = wrapper.find('nuxt-link-stub')
      expect(link.attributes().to).toBe('/')
    })

    test('img should have a src as logo img', () => {
      const img = wrapper.find('img')
      expect(img.attributes().src).toBe('~assets/logo.svg')
    })

    test('img should have a alt as logo', () => {
      const img = wrapper.find('img')
      expect(img.attributes().alt).toBe('logo')
    })

    test('img should have a logo class', () => {
      const img = wrapper.find('img')
      expect(img.classes()).toContain('logo')
    })
  })

  describe('snapshot tests', () => {
    test('should match the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
