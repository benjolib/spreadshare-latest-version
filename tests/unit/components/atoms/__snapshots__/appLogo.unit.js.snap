// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appLogo snapshot tests should match the snapshot 1`] = `
<nuxt-link-stub
  to="/"
>
  <img
    alt="logo"
    class="logo"
    src="~assets/logo.svg"
  />
</nuxt-link-stub>
`;
